nikola/plugins/compile/markdown/__init__.py

```diff
@@ -153,6 +157,12 @@ def read_metadata(self, post, file_metadata_regexp=None, unslugify_titles=False,
         source = post.translated_source_path(lang)
         with io.open(source, 'r', encoding='utf-8') as inf:
             # Note: markdown meta returns lowercase keys
+            data = inf.read()
+            # If the metadata starts with "---" it's actually YAML and
+            # we should not let markdown parse it, because it will do
+            # bad things like setting empty tags to "''"
+            if data.splitlines()[0] == '---':
```

Comment:
What’s the behavior with `+++` ? also, you can just check `if data.startswith('---\n'):` (Windows-proof because of `r` mode)